src/traces/contour/plot.js

```diff
@@ -259,11 +264,28 @@ function joinAllPaths(pi, perimeter) {
     return fullpath;
 }
 
-function makeLines(plotgroup, pathinfo, contours) {
+var TRAILING_ZEROS = /\.?0+$/;
+
+function makeLinesAndLabels(plotgroup, pathinfo, gd, cd0, contours, perimeter) {
```

Comment:
Except for the innermost join content, it looks like mostly d3 operations so is probably okay, but this function comes out to 240 lines, which seems a bit on the high side.